app/views/granules/descriptor_document.xml.erb

```diff
@@ -15,8 +15,12 @@
 	<os:Contact><%= ENV['contact']%></os:Contact>
 	<os:Url type="application/atom+xml" rel="results"
 	  params:method="GET"
-	  template="<%= ENV['opensearch_url']%>/granules.atom?datasetId={echo:datasetId?}&amp;shortName=<%= @short_name %>&amp;versionId=<%= @version_id %>&amp;dataCenter=<%= @data_center %>&amp;boundingBox={geo:box?}&amp;lat={geo:lat?}&amp;lon={geo:lon?}&amp;radius={geo:radius?}&amp;geometry={geo:geometry?}&amp;placeName={geo:name?}&amp;startTime={time:start?}&amp;endTime={time:end?}&amp;cursor={os:startPage?}&amp;numberOfResults={os:count?}&amp;offset={os:startIndex?}&amp;uid={geo:uid?}&amp;parentIdentifier={eo:parentIdentifier?}&amp;clientId=<%= @client_id_model.clientId %>">
-	  <params:Parameter name="datasetId" uiDisplay="Collection identifier" value="{echo:datasetId}" title="Inventory associated with a dataset expressed as an ID" minimum="0"/>
+		<% if !@client_id_model.clientId.nil? && !@client_id_model.clientId.blank? -%>
```

Comment:
Again, I think this should be abstracted behind the model validation.

Reply:
There are two points which need to be reconciled here:

First, by way of changes made in this PR, no clientId is now a valid clientId.
Second, we only want to include clientId in the template if one is supplied.

To reconcile this, the blank? check is necessary. However, the nil? check was abstracted behind the clientId model, so we now check for valid? instead.